File: 179-largest-number/179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        import functools
        def comparator(s1, s2):
            if s1+s2 < s2+s1:
                return -1
            if s1+s2 > s2+s1:
                return 1
            return 0
        
        nums = [str(num) for num in nums]
        nums = sorted(nums, key = functools.cmp_to_key(comparator),  reverse = True)
        res = '0' if nums[0] == '0' else ''.join(nums)      
        return res
        

File: 179-largest-number/179-largest-number.java
class Solution {
    

public String largestNumber(int[] nums) {
    if (nums == null || nums.length == 0) {
        return "";
    }
    String[] largest = new String[nums.length];
    for (int i = 0; i < nums.length; i++) {
        largest[i] = String.valueOf(nums[i]);
    }
    Arrays.sort(largest, (a, b) -> (b + a).compareTo(a + b));
    
    if (largest[0].equals("0")) {
         return "0";
    }
    StringBuilder res = new StringBuilder();
    for (String str : largest) {
        res.append(str);
    }
    return res.toString();
        
 }
}
